src/Stryker.Core/Stryker.Core/TestRunners/VsTest/VsTestRunnerFactory.cs

```diff
@@ -0,0 +1,56 @@
+﻿using Microsoft.VisualStudio.TestPlatform.ObjectModel;
+using Stryker.Core.Initialisation;
+using Stryker.Core.Options;
+using System.Collections.Concurrent;
+using System.Collections.Generic;
+
+namespace Stryker.Core.TestRunners.VsTest
+{
+    public class VsTestRunnerFactory : ITestRunner
+    {
+        private readonly ConcurrentBag<VsTestRunner> _availableRunners = new ConcurrentBag<VsTestRunner>();
+
+        public VsTestRunnerFactory(StrykerOptions options, ProjectInfo projectInfo)
+        {
+            IEnumerable<TestCase> testCases;
+
+            using (var runner = new VsTestRunner(options, projectInfo, null))
+            {
+                testCases = runner.DiscoverTests();
+            }
+
+            for (var i = 0; i < options.ConcurrentTestrunners; i++)
+            {
+                _availableRunners.Add(new VsTestRunner(options, projectInfo, testCases));
+            }
+        }
+
+        public TestRunResult RunAll(int? timeoutMS, int? activeMutationId)
+        {
+            TestRunResult result = null;
+            while (result is null)
```

Comment:
Is this busy waiting needed? Can't we solve this with event handlers? Like, set a handler when an object is free again and wait here till the object is free.